MapStory/mapstory
# 939 sort api

Issue description: The ordering / sorting methods on the explore page were front-end based and didn't affect what content was being pulled from the API. In local and test environs, since there weren't more than 1 page of results, these methods appeared to work, as they sorted a single page worth of results into the anticipated order. It came to my attention during the "featured cards" work that these methods were doing nothing to sort results _before_ they reach the browser and therefore were ineffective to sort content across multiple pages of results. Closes #939 

TLDR: we are now sorting results server side rather than client side :)

## To test:

- Change your results-per-page limit [here](https://github.com/MapStory/mapstory/blob/master/mapstory/static/mapstory/js/src/search/explore.controller.js#L16) to `2` or `3` 
    - make sure you run `grunt concat` from `mapstory/static` anytime you change js (or `grunt watch`)
    - alternatively you can do this [here](https://github.com/MapStory/mapstory/blob/53efb615668b9cccfdf403c6ea995543a1490b63/mapstory/settings/base.py#L546) in the base settings, then do whatever to have an instance running with that updated setting

##### Content Sorting: 

- Have some content (layers and/or stories) with view counts 
    - either modified via the admin panel [layers](https://docker/admin/layers/layer/) `Popular count:` field.
    - or racked up via viewing a detail page from a non-admin, non-resource owner account  

- Verify that your results are sorted (across pages) in highest to lowest view count upon landing on the page, AND upon selecting 'Popular' from the dropdown menu on the right, 
- Verify that your results update and are sorted from Newest to Oldest upon selecting 'Newest' from the dropdown menu on the right


##### StoryTeller Sorting:
- You should still have 2 or 3 results per-page from the step above
- Have or create multiple user accounts
- Select the "StoryTellers" Tab at the top of the explore page
- Verify that your results are sorted A-Z by Username upon landing on the page AND upon selecting 'Username A-Z' from the dropdown on the right
- Verify that your results are updated and sorted Z-A by username upon selecting 'Username Z-A' from the dropdown on the right. 


Go ahead and change your search limit back, or just checkout the file. 
Thanks!
